[twig mode] Add `base` option support for mixed mode

TDaglis opened the pull request:
Updated Twig mode to an multiplexingMode.
Twig mode now supports a base option and the base is properly highlighted.

#### Examples
##### html base
before:
![before](https://cloud.githubusercontent.com/assets/3427236/13455650/715e35c6-e067-11e5-9ba3-e911309a9e7b.PNG)
after:
![after](https://cloud.githubusercontent.com/assets/3427236/13455654/730f9c8e-e067-11e5-93c1-eee5a7e3f953.PNG)

##### xml base
before: 
![image](https://cloud.githubusercontent.com/assets/3427236/13461328/640db154-e089-11e5-8f47-4c950212738a.png)
after:
![beforexml](https://cloud.githubusercontent.com/assets/3427236/13461331/6a2819c6-e089-11e5-9721-9d6dd95b2075.PNG)

marijnh commented:
Would it ever make sense to use Twig with a non-HTML output language? If so, could you adjust your implementation to support a `base` option to configure the base language, the way the Handlebars mode does?

TDaglis commented:
Even though HTML would be the typical use case, it would make sense to use Twig with a non-HTML base. I updated the PR accordingly to support a `base` option as you suggested :+1:

marijnh commented:
(Is the fact that you're now defaulting to no base mode intentional?)

TDaglis commented:
Yes. For backwards compatibility mostly and in order to make the mode more generic.
I also just fixed the delimiters of the last commit since they did not work properly. I replaced them with regexes.

marijnh commented:
Thanks. Merged as a448b97
